cmd/agent/app/snmpwalk.go

```diff
@@ -0,0 +1,316 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gosnmp/gosnmp"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultVersion = ""
+	defaultOID     = ""
+	defaultPort    = 161
+
+	// snmp v1 & v2c
+	defaultCommunityString = "public"
+
+	// snmp v3
+	defaultUserName      = ""
+	defaultAuthProtocol  = ""
+	defaultAuthKey       = ""
+	defaultPrivProtocol  = ""
+	defaultPrivKey       = ""
+	defaultContext       = "public"
+	defaultSecurityLevel = ""
+
+	// general communication options
+	defaultTimeout = 10 // Timeout better suited to walking
+	defaultRetries = 3
+)
+
+var (
+	// general
+	snmpVersion string
+	address     string
+	deviceIP    string
+	oid         string
+	port        uint16
+	value       uint64
+	setVersion  gosnmp.SnmpVersion
+	snmp        gosnmp.GoSNMP
+
+	// v1 & v2c
+	communityString string
+
+	// v3
+	user          string
+	authProtocol  gosnmp.SnmpV3AuthProtocol
+	authProt      string
+	authKey       string
+	privProtocol  gosnmp.SnmpV3PrivProtocol
+	privProt      string
+	privKey       string
+	snmpContext   string
+	msgFlags      gosnmp.SnmpV3MsgFlags
+	securityLevel string
+
+	// communication
+	retries int
+	timeout int
+)
+
+var (
+	snmpCmd = &cobra.Command{
+		Use:   "snmp",
+		Short: "Snmp tools",
+		Long:  ``,
+	}
+)
+
+func init() {
+
+	snmpWalkCmd.Flags().StringVarP(&snmpVersion, "snmp-version", "v", defaultVersion, "Specify SNMP version to use")
```

Comment:
I think this is opinionated. It has some pros to have dedicated variables, we can easily change default this way. I don't see this change as a blocker, do you agree @AlexandreYang ?

Reply:
Yes, not blocker :)

For empty defaults, I think inlining is more readable since we don't need to go the the default definition to see the value. Using a default variable also suggest there is a "meaningful" default value, which is not the case for empty value.